### PR TITLE
add InvalidMessageException tests

### PR DESCRIPTION
**Pull request by uweenukr**

### CODE REVIEW

File: Assets/Mirror/Tests/Editor/InvalidMessageExceptionTest.cs [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    public class InvalidMessageExceptionTest
+    {
+        [Test]
+        public void InvalidMessageTest()
+        {
+            ExceptionTest exceptionTest = new ExceptionTest();
+
+            InvalidMessageException ex = Assert.Throws<InvalidMessageException>(() =>
+            {
+                exceptionTest.ThrowsOne();
```
Comment (paulpach) [inline Assets/Mirror/Tests/Editor/InvalidMessageExceptionTest.cs:33]:
```suggestion
                throw new InvalidMessageException();
```

File: Assets/Mirror/Tests/Editor/InvalidMessageExceptionTest.cs [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+            });
+
+            ex = Assert.Throws<InvalidMessageException>(() =>
+            {
+                exceptionTest.ThrowsTwo();
```
Comment (paulpach) [inline Assets/Mirror/Tests/Editor/InvalidMessageExceptionTest.cs:38]:
```suggestion
                throw new InvalidMessageException("Test Message");
```

File: Assets/Mirror/Tests/Editor/InvalidMessageExceptionTest.cs [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+            });
+
+            Assert.That(ex.Message, Is.EqualTo("Test Message"));
+
+            ex = Assert.Throws<InvalidMessageException>(() =>
+            {
+                exceptionTest.ThrowsThree();
```
Comment (paulpach) [inline Assets/Mirror/Tests/Editor/InvalidMessageExceptionTest.cs:45]:
```suggestion
                throw new InvalidMessageException("Test Message Too", new System.Exception());
```

File: Assets/Mirror/Tests/Editor/InvalidMessageExceptionTest.cs [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+            });
+
+            Assert.That(ex.Message, Is.EqualTo("Test Message Too"));
+            Assert.That(ex.InnerException, Is.TypeOf(typeof(System.Exception)));
```
Comment (paulpach) [inline Assets/Mirror/Tests/Editor/InvalidMessageExceptionTest.cs:49]: no need to,  you would not be able to throw it if it wasn't an Exception
```suggestion
```

Comment (uweenukr) [reply Assets/Mirror/Tests/Editor/InvalidMessageExceptionTest.cs:49]: How would I test that the inner exception is there without asserting for it?

**Comment by uweenukr** [comment]
All but the last suggestion, that i had a question about, was applied.

**Pull request merged**